### DISCUSSION TITLE
Trust the system root CA certificates

### DISCUSSION THREAD

**Discussion by deltoura** [npm/feedback#497] [CLI] [answered]
**Problem Statement**

As explained in https://npm.community/t/trust-the-system-root-ca-certificates/6429:

Lots of organizations have a system trust store with custom root CA certificates, for trusting internal repositories.
With npm you need to use the `cafile` option to be able to use them.
This is not very convenient:

- you need to add this additional configuration
- you add a strong dependency to the name of your certificate file
- you need to duplicate the entire trust store if you need to trust anything else than your internal sources

This can be blocker for some organizations.

**Potential Solution**

npm should automatically trust the root CA certificates in system trust store.
This is usually the case for other tools (e.g. Java, Maven, Sbt, ...)

**Comment by ljharb** [comment]
The downside i can think of is that it makes a project no longer portable - in other words, every developer on the same project would suddenly no longer have access to the same certificates, which might mean they can’t all install deps equally well.

**Reply by deltoura** [reply]
I am not sure I understand your point - this is already the case if developers have a different `cafile` option in their local npm configuration?

In an organization with an organization-wide CA configuration, the projects would be portable inside the organization.
Projects that need to be universally portable should use only dependencies from repositories trusted by standard CA certificates.

**Reply by ljharb** [reply]
Yes, if the cafile path is located outside the project repo, it wouldn't be portable - but i'd assume it would be located in the repo, for portability reasons.

> In an organization with an organization-wide CA configuration, the projects would be portable inside the organization.
Projects that need to be universally portable should use only dependencies from repositories trusted by standard CA certificates.

Fair point, I completely agree with this.

**Comment by MylesBorins** [comment, marked as answer]
@deltoura a next step if you are interested in seeing the npm CLI support this feature would be to open an RFC issue at https://github.com/npm/rfcs

**Reply by deltoura** [reply]
OK I will do that then. Thanks.

**Reply by deltoura** [reply]
See https://github.com/npm/rfcs/pull/418
